neo/rawio/plexon2rawio/plexon2rawio.py

```diff
@@ -0,0 +1,494 @@
+"""
+Plexon2RawIO is a class to read Plexon PL2 files.
+
+On non-windows systems this IO requires Wine, see https://www.winehq.org/
+
+Note on IO performance and memory consumption:
+This IO is only partially lazy: To load a data chunk of an individual
+signal channels, the complete data of that channel will be loaded into memory
+and cached. This occurs as a limitation of the pypl2 package and might be improved
+in future versions by exploiting the block-wise loading of signals.
+
+To clear the currently cached signal data use the `clear_analogsignal_cache()` method.
+
+Note on Neo header and spike channels:
+The IO only considers enabled channels and will not list disabled channels in its header.
+
+There is no 1-1 correspondence of PL2 spike channels and neo spike channels as each unit of
+a PL2 spike channel will be represented as an individual neo spike channel.
+
+Author: Julia Sprenger
+"""
+import pathlib
+from collections import namedtuple
+from urllib.request import urlopen
+from datetime import datetime
+
+import numpy as np
+
+
+from ..baserawio import (BaseRawIO, _signal_channel_dtype, _signal_stream_dtype,
+                         _spike_channel_dtype, _event_channel_dtype)
+
+
+class Plexon2RawIO(BaseRawIO):
+    """
+    Class for "reading" data from a PL2 file
+
+    Usage:
+        >>> import neo.rawio
+        >>> r = neo.rawio.Plexon2RawIO(filename='my_data.pl2')
+        >>> r.parse_header()
+        >>> print(r)
+        >>> raw_chunk = r.get_analogsignal_chunk(block_index=0, seg_index=0,
+                            i_start=0, i_stop=1024,  stream_index=0, channel_indexes=range(10))
+        >>> float_chunk = r.rescale_signal_raw_to_float(raw_chunk, dtype='float64',
+                            channel_indexes=[0, 3, 6])
+        >>> spike_timestamp = r.get_spike_timestamps(spike_channel_index=0, t_start=None,
+                            t_stop=None)
+        >>> spike_times = r.rescale_spike_timestamp(spike_timestamp, 'float64')
+        >>> ev_timestamps, _, ev_labels = r.get_event_timestamps(event_channel_index=0)
+
+    """
+    extensions = ['pl2']
+    rawmode = 'one-file'
+
+    def __init__(self, filename='', pl2_dll_file_path=None):
+
+        # signals, event and spiking data will be cached
+        # cached signal data can be cleared using `clear_analogsignal_cache()()`
+        self._analogsignal_cache = {}
+        self._event_channel_cache = {}
+        self._spike_channel_cache = {}
+
+        # note that this filename is used in self._source_name
+        self.filename = pathlib.Path(filename)
+
+        if (not self.filename.exists()) or (not self.filename.is_file()):
+            raise ValueError(f'{self.filename} is not a file.')
+
+        BaseRawIO.__init__(self)
+
+        # download default PL2 dll once if not yet available
+        if pl2_dll_file_path is None:
+            pl2_dll_folder = pathlib.Path .home() / '.plexon_dlls_for_neo'
```

Comment:
Maybe this should live in the package directory. That way when you uninstall the package you uninstall the dll

Reply:
This approach might cause problems if the installation was performed with different right than the execution. The package directory might be read-only. Creating a folder for the dll in the home directory should not fail due to permissions and is the same location used also for cloning of the `ephy_testing_files` repository. I agree that there should be a mechanism to register these additional folders in the pyproject.toml to be also removed upon deinstallation of neo, but didn't find the right option yet. Let me know if you have seen something like this already.